python/lsst/meas/deblender/plugins.py

```diff
@@ -166,111 +167,122 @@ def buildMultibandTemplates(debResult, log, useWeights=False, usePsf=False,
-        if "l0_thresh" in multibandKwargs:
-            kwargs["l0_thresh"] = multibandKwargs["l0_thresh"]
-            del multibandKwargs["l0_thresh"]
-        elif "l1_thresh" in multibandKwargs:
-            kwargs["l1_thresh"] = multibandKwargs["l1_thresh"]
-            del multibandKwargs["l1_thresh"]
-        multibandKwargs["prox_S"] = deblender.proximal.strict_monotonicity(data, peaks=peaks, **kwargs)
-    # Update the morphology before the SED in each step, since the initial SED is better constrained
-    if "update_order" not in multibandKwargs:
-        multibandKwargs["update_order"] = [1,0]
-
-    if "psfs" in multibandKwargs:
-        log.warn("PSF convolution has not been optimized and is currently very slow")
+        psf = np.array(psfs)
+    else:
+        psf = None
+
+    bg_rms = np.array([debResult.deblendedParents[f].avgNoise for f in debResult.filters])*bg_scale
+    if sources is None:
+        # If only a single constraint was given, use it for all of the sources
+        if (constraints is scarlet.constraints.Constraint or
+            constraints is scarlet.constraints.ConstraintList
+        ):
+            constraints = [constraints.copy() for peak in peaks]
+        elif constraints is None:
+            constraints = [None]*len(peaks)
+        sources = [
+            scarlet.source.ExtendedSource(center=peak,
+                                          img=data,
+                                          bg_rms=bg_rms,
+                                          constraints=constraints[pk],
+                                          psf=psf,
+                                          symmetric=True,
+                                          monotonic=True,
+                                          thresh=1.0,
+                                          config=config)
+            for pk,peak in enumerate(peaks)
+        ]
 
     # When a footprint includes only non-detections
     # (peaks in the noise too low to deblend as a source)
     # the deblender currently fails.
     try:
-        result = deblender.nmf.deblend(img=data, peaks=peaks, **multibandKwargs)
+        blend = scarlet.blend.Blend(sources=sources, img=data, weights=weights, bg_rms=bg_rms, config=config)
+        blend.fit(steps=maxIter, e_rel=relErr)
     except np.linalg.LinAlgError as e:
         log.warn("Deblend failed catastrophically, most likely due to no signal in the footprint")
         debResult.failed = True
         return False
-    debResult.multibandResult = result
+    debResult.blend = blend
 
     modified = False
     # Create the Templates for each peak in each filter
-    for pk, peak in enumerate(result.T.peaks.peaks):
+    for pk, src in enumerate(blend.sources):
+        _cx = src.Nx >> 1
+        _cy = src.Ny >> 1
+
         if debResult.peaks[pk].skip:
             continue
         modified = True
-        cx = peak.x+xmin
-        cy = peak.y+ymin
+        cx = src.center[1]+xmin
+        cy = src.center[0]+ymin
```

Comment:
(y, x)? Just checking ;)

Reply:
yes, unfortunately I begrudgingly agreed to (y,x) pixel coordinates